visivo/commands/deploy_phase.py

```diff
@@ -1,19 +1,117 @@
 import click
 import requests
 import json
+import os
+import asyncio
+import aiofiles
+import httpx
+from time import time
+from tenacity import retry, stop_after_attempt, wait_fixed
 from visivo.commands.utils import get_profile_file, get_profile_token
 from visivo.discovery.discover import Discover
 from visivo.logging.logger import Logger
 from visivo.parsers.serializer import Serializer
 from visivo.parsers.parser_factory import ParserFactory
 
+# Limit concurrent uploads to avoid overloading the API
+semaphore = asyncio.Semaphore(50)
+
+@retry(stop=stop_after_attempt(3), wait=wait_fixed(2))
```

Comment:
I would really love a test for these functions, they don't have to be exhaustive, but I also don't want to accidentally break them.

Reply:
Yeah that's a good point. I'll add some tests for these